### PR TITLE
Switch from nodemon to pm2

### PR DESCRIPTION
**Pull request by netes**
## Description
This PR replaces `nodemon` with `PM2`.

## Reasoning
Reasons for this are pretty straightforward - PM2 is much more powerful tool with variety of configuration options, built-in metrics, scaling and Docker support.

### Starting the app
I've made some modification to `npm scripts`, right now we've got 3 starting scripts:
- `start` (development and debugging - which replaces old `start` and `debug`)
- `start:prod` (stg/prod without Docker)
- `start:prod-docker` (stg/prod with Docker).

### PM2 Configuration
PM2 configuration files are kept in `pm2-process-dev.json` and `pm2-process-prod.json`.

### Optional Keymetrics
There are additional 2 config variables in `.env` - `KEYMETRICS_PUBLIC` and `KEYMETRICS_SECRET`. By filling them app will be automatically connected [Keymetrics](https://app.keymetrics.io/) monitoring tool.

@RafalWilinski, @mrowa44, @bardzusny, @jniechcial WDYT guys?

### CODE REVIEW

File: pm2-process-prod.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "name": "ng-node-starter-app",
+  "script": "./app.js",
```
Comment (RafalWilinski) [inline pm2-process-prod.json]:
```js
"exec_mode": "cluster",
"instances": 4 // ?
```

Comment (netes) [inline pm2-process-prod.json]: I think we shouldn't set it by default. If anyone will need clustering, setting it is effortless.

**Comment by mrowa44** [comment]
pm2 config files can be also in js format, is there a reason we use json? I think generally js is easier to manage and json can be cumbersome. We have eslint config in js too. That's just a suggestion though. Everything else looks good 👍

File: .env.development.example [hunk 2]
```diff
@@ -4,3 +4,5 @@ DB_DATABASE=database_development
 DB_HOST=127.0.0.1
 JWT_TOKEN=secret
 TOKEN_EXPIRATION_TIME=604800000
+KEYMETRICS_PUBLIC=
+KEYMETRICS_SECRET=
```
Comment (bardzusny) [inline .env.development.example]: I think this is something to revert (remove) if we end up not exploiting Keymetrics in most real-life projects - just a note.

Comment (netes) [inline .env.development.example]: We can discuss it internally 👍

**Pull request merged**